### DISCUSSION TITLE
How should an `at://` URI with a handle be resolved?

### DISCUSSION THREAD

**Discussion by snarfed** [bluesky-social/atproto#1591] [Protocol (atproto)]
Hi all! Came across an `at://` URI quirk today: in a future federated world, if I have an `at://` URI with a handle, eg `at://bnewbold.bsky.team/app.bsky.feed.post/3jwdwj2ctlk26`, how can I find the PDS to use to resolve its handle to a DID?

If I already have a DID document with the handle stored locally, I can look that up. But if I don't, I can't know where to call `resolveHandle`, right?

Should PLC directories eventually implement `resolveHandle` themselves to support this? Or am I expected to never encounter or handle "bare" handle-based `at://` URIs like this without already having resolved their DID?

**Comment by snarfed** [comment]
Oh, never mind, this is handle resolution, https://atproto.com/specs/handle#handle-resolution . Hah, ok!

**Reply by bnewbold** [reply]
yup yup!
